dask_expr/io/io.py

```diff
@@ -199,12 +205,18 @@ def _simplify_up(self, parent):
     def _divisions(self):
         return self._divisions_and_locations[0]
 
+    @functools.cached_property
+    def npartitions(self):
+        if self._filtered:
+            return super().npartitions
+        return len(self._divisions_and_locations[0]) - 1
+
     def _locations(self):
         return self._divisions_and_locations[1]
 
     def _filtered_task(self, index: int):
         start, stop = self._locations()[index : index + 2]
-        part = self.frame.iloc[start:stop]
+        part = self._sorted_data.iloc[start:stop]
```

Comment:
I don't think we want to do this unless `self.sort` is `True`?

Reply:
Oh yes, thx for catching. Added a test as well...